sw/device/tests/aon_timer_wakeup_irq_test.c

```diff
@@ -0,0 +1,126 @@
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+static const uint32_t kAonTimerFreq = 200 * 1000;  // 200kHz
+static const uint32_t kSleepTime_us = 2 * 1000;    // 2ms
+static dif_aon_timer_t aon_timer;
+static dif_rv_plic_t plic;
+static dif_rv_plic_irq_id_t plic_irq_id;
+static ibex_timeout_t timeout;
+
+static void aon_timer_wakeup_config(dif_aon_timer_t *aon_timer,
+                                    uint32_t wakeup_threshold) {
+  // Make sure that wake-up timer is stopped.
+  CHECK_DIF_OK(dif_aon_timer_wakeup_stop(aon_timer));
+
+  // Make sure the wake-up IRQ is cleared to avoid false positive.
+  CHECK_DIF_OK(dif_aon_timer_irq_acknowledge(aon_timer,
+                                             kDifAonTimerIrqWkupTimerExpired));
+
+  // Check whether the irq is pending
+  bool is_pending;
+  CHECK_DIF_OK(dif_aon_timer_irq_is_pending(
+      aon_timer, kDifAonTimerIrqWkupTimerExpired, &is_pending));
+  CHECK(!is_pending);
+
+  // Start timer
+  CHECK_DIF_OK(dif_aon_timer_wakeup_start(aon_timer, wakeup_threshold, 0));
+}
+
+static void execute_test(dif_aon_timer_t *aon_timer) {
+  plic_irq_id = kTopEarlgreyPlicPeripheralUnknown;
+
+  LOG_INFO("Wainting wakeup interruption");
+
+  // Capture the current tick to measure the time the irq will take
+  timeout = ibex_timeout_init(kSleepTime_us * 1000);
+
+  // Setup the timer and wait the irq
+  uint32_t sleep_cycles = (kSleepTime_us * kAonTimerFreq / 1000000);
+  aon_timer_wakeup_config(aon_timer, sleep_cycles);
+  wait_for_interrupt();
+}
+
+void ottf_external_isr(void) {
+  // Calc the elapsed time since the activation of the irq
+  uint32_t time_elapsed = ibex_timeout_elapsed(&timeout);
+
+  CHECK_DIF_OK(
+      dif_rv_plic_irq_claim(&plic, kTopEarlgreyPlicTargetIbex0, &plic_irq_id));
+
+  dif_aon_timer_irq_t irq = (dif_aon_timer_irq_t)(
+      plic_irq_id -
+      (dif_rv_plic_irq_id_t)kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralAonTimerAon,
+        "Interrupt from incorrect peripheral: exp = Aon Timer, obs = %d",
+        peripheral);
+
+  CHECK(irq == kDifAonTimerIrqWkupTimerExpired,
+        "Interrupt type incorrect: exp = wakeupTimer, obs = %d", irq);
+
+  CHECK(time_elapsed >= kSleepTime_us,
+        "Timer took %u usec which is less than %u", time_elapsed,
+        kSleepTime_us);
+
+  CHECK_DIF_OK(dif_aon_timer_irq_acknowledge(&aon_timer, irq));
+}
+
+bool test_main(void) {
+  LOG_INFO("Starting aon timer wakeup interrupt test");
```

Comment:
nit: you can remove this. [Our test framework indicates when the test is beginning.](https://github.com/lowRISC/opentitan/blob/616eadc1a1133ac45421f32ecc8d4458ca1daf4c/sw/device/lib/testing/test_framework/ottf.c#L65
)